src/rust/engine/logging/src/lib.rs

```diff
@@ -89,3 +89,7 @@ impl From<PythonLogLevel> for log::Level {
     }
   }
 }
+
+mod pants_packages {
+  include!(concat!(env!("OUT_DIR"), "/packages.rs"));
```

Comment:
Is this better than writing the names of the packages to a .txt file and reading normally?
The documentation for `include!` seems to hint that it's unrecommended to use it at all.
I guess this is precomputed at compile time, and would be better than something like a `lazy_static!` to save the overhead, so I'm okay with it, just wondering about your thoughts there.

Reply:
Yeah; if we wanted to read a text file at runtime, we would need to distribute that text file alongside pants somewhere, and then we'd need to know how to locate it on disk, and then we'd need to read it from disk every time pants runs. All of that adds a lot of complexity vs just knowing this information at compile time.

The advice against using `include!` in https://doc.rust-lang.org/std/macro.include.html is mostly talking about concerns around scoping; in this specific example, the only symbol we're relying on being exposed to our snippet is `str`, so the concern is that if in `logging/lib.rs` we shadowed the `str` symbol, that shadowed version would be used (and our code probably wouldn't compile). Accordingly, technically the codegen would be more correct if it used `::std::str` rather than `str` - I will make this change!

When you're basically generating a whole module, `include!` is generally fine, but it's much sketchier to use for smaller chunks of code which may be more sensitive to the surrounding context.

The other down-side of `include!` is that IDEs tend not to click through to it nicely; hopefully that will get resolved at some point...

Reply:
... Apparently you can't absolutely qualify `str` (in any way that I can find), so... Leaving as-is...